skiko/src/commonTest/kotlin/org/jetbrains/skia/TextLineTest.kt

```diff
@@ -125,22 +123,108 @@ class TextLineTest {
             assertCloseEnough(41f, line.getCoordAtOffset(4))
         }
 
+        assertEquals("ă", ManagedString("ă").toString())
+        assertEquals("aa̧", ManagedString("aa̧").toString())
+
         TextLine.make("ă", jbMono36()).use { line -> }
         TextLine.make("aa̧", jbMono36()).use { line ->
+            assertEquals(42, line.glyphs[0])
             // JetBrains Mono supports “a” but not “ ̧ ”
             // Second grapheme cluster should fall back together, second “a” should resolve to different glyph
-            assertNotEquals(line.glyphs.get(0), line.glyphs.get(1))
+            if (kotlinBackend.isNotJs()) { // TODO(karpovich): figure out why js gives different result
+                assertNotEquals(line.glyphs[0], line.glyphs[1])
+            }
         }
     }
 
     @Test
     fun emojiTest() = runTest {
+        assertEquals("☺", ManagedString("☺").toString())
+        assertEquals("☺️", ManagedString("☺️").toString())
+
         TextLine.make("☺", firaCode36()).use { misc ->
             TextLine.make("☺️", firaCode36()).use { emoji ->
                 assertContentEquals(shortArrayOf(1706), misc.glyphs)
                 assertEquals(1, emoji.glyphs.size)
-                assertNotEquals(misc.glyphs.get(0), emoji.glyphs.get(0))
+                if (kotlinBackend.isNotJs()) { // TODO(karpovich): figure out why js gives different result
```

Comment:
I'm not sure that emojis are present in Fira Code's glyph set. In this case emojis may be rendered from another font

Reply:
Right, font callbacks (no available on JS) could be the reason why emojis are properly rendered on those platforms. Maybe execute tests with special font manager not having system fallbacks?

Reply:
Or at least add TODO for that.

Reply:
Unfortunately NotoEmoji didn't help - both glyphs were the same.  So added TODO for now.

P.S.
Interesting:
<img width="800" alt="Screenshot 2021-11-08 at 10 50 51" src="https://user-images.githubusercontent.com/7372778/140720631-8eaff587-fbe5-4bbf-9ec5-0969af40d652.png">

Github ui also treats these both emojis equally, although they're different in IDE.